tss-esapi/src/structures/attestation.rs

```diff
@@ -0,0 +1,261 @@
+// Copyright 2021 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use crate::constants::tss::{
+    TPM2_ST_ATTEST_CERTIFY, TPM2_ST_ATTEST_COMMAND_AUDIT, TPM2_ST_ATTEST_CREATION,
+    TPM2_ST_ATTEST_NV, TPM2_ST_ATTEST_NV_DIGEST, TPM2_ST_ATTEST_QUOTE,
+    TPM2_ST_ATTEST_SESSION_AUDIT, TPM2_ST_ATTEST_TIME,
+};
+use crate::constants::TPM_GENERATED_VALUE;
+use crate::structures::{ClockInfo, Data, Name};
+use crate::tss2_esys::{
+    Tss2_MU_TPMS_ATTEST_Unmarshal, TPM2B_ATTEST, TPMI_ST_ATTEST, TPMS_ATTEST, TPMS_CERTIFY_INFO,
+    TPMU_ATTEST,
+};
+use crate::{Error, Result, WrapperErrorKind};
+use log::error;
+use std::convert::{TryFrom, TryInto};
+use std::ops::Deref;
+use zeroize::Zeroizing;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Attest {
+    attestation_type: AttestationType,
```

Comment:
Should I keep this field? It seems superfluous, since `attested` below will have to be consistent with it anyway. I think it'd be cleaner to just keep `attested` and create a mapping from that.